### PR TITLE
add embedded kafka and zookeeper servers for UTs

### PR DESCRIPTION
**Pull request by eambrosio**
## Description

Use embedded Kafka and Zookeeper servers for UTs #58 

### Testing
- [X] Unit, integration, acceptance tests as appropriate
- [ ] Coverage (>90%)

### Documentation
- [ ] Changelog update
- [ ] Documentation entry

### Scalastyle
Result of scalastyle execution:

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -159,30 +152,17 @@
...
         </dependency>
         <dependency>
-            <groupId>org.scalacheck</groupId>
-            <artifactId>scalacheck_${scala.binary.version}</artifactId>
-            <version>${scalacheck.version}</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
             <groupId>com.typesafe.play</groupId>
             <artifactId>twirl-api_2.11</artifactId>
```
Comment (eruizgar) [inline pom.xml]: Could you change by twirl-api_${scala.binary.version}?

File: pom.xml [hunk 2]
```diff
@@ -222,11 +202,72 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.apache.kafka</groupId>
+            <artifactId>kafka-clients</artifactId>
+            <version>${kafka.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.typesafe.scala-logging</groupId>
+            <artifactId>scala-logging_${scala.binary.version}</artifactId>
+            <version>3.5.0</version>
```
Comment (eruizgar) [inline pom.xml]: Could you use a variable to specify version?

File: pom.xml [hunk 3]
```diff
@@ -222,11 +202,72 @@
...
+            <artifactId>kafka-clients</artifactId>
+            <version>${kafka.version}</version>
+            <classifier>test</classifier>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
             <groupId>org.apache.curator</groupId>
             <artifactId>curator-test</artifactId>
-            <version>${curator-framework.version}</version>
+            <version>2.9.0</version>
```
Comment (eruizgar) [inline pom.xml]: Same as above

File: pom.xml [hunk 4]
```diff
@@ -222,11 +202,72 @@
...
+        <dependency>
+            <groupId>org.scalacheck</groupId>
+            <artifactId>scalacheck_${scala.binary.version}</artifactId>
+            <version>${scalacheck.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <version>3.3.0</version>
```
Comment (eruizgar) [inline pom.xml]: Same as above

**Review by eruizgar — CHANGES_REQUESTED** [review]

**Review by eruizgar — APPROVED** [review]

**Review by albertostratio — APPROVED** [review]
LGTM

**Pull request merged**